pkg/controller/infrastructure/infraflow/reconcile.go

```diff
@@ -522,22 +568,46 @@ func (c *FlowContext) ensureZones(ctx context.Context) error {
 		tagsPrivate[TagKeyRolePrivateELB] = TagValueUse
 		desired = append(desired,
 			&awsclient.Subnet{
-				Tags:             tagsWorkers,
-				VpcId:            c.state.Get(IdentifierVPC),
-				CidrBlock:        zone.Workers,
-				AvailabilityZone: zone.Name,
+				Tags:      tagsWorkers,
+				VpcId:     c.state.Get(IdentifierVPC),
+				CidrBlock: zone.Workers,
+				Ipv6CidrBlocks: func() []string {
+					if subnetCIDR1 != "" {
+						return []string{subnetCIDR1}
+					} else {
+						return nil
+					}
```

Comment:
If there was an array of type `[]*string` we could get rid of this function ugliness...